dsymbol/src/dsymbol/conversion/first.d

```diff
@@ -944,10 +946,23 @@ private:
 		}
 	}
 
+	void setFunctionReturnType(const Type returnType, ref DSymbol acSymbol){
+		if (returnType !is null)
+		{
+			auto app = appender!string();
+			app.formatNode(returnType);
+			auto foundSymbols = currentScope.getSymbolsByName(istring(app.data));
```

Comment:
looking up symbols in the symbol cache by name might not be that accurate and will hinder us from implementing things like UFCS on templates. I think the parameters/return type processing of functions might need to be put into a separate semantic step. Also I think imports might not be resolved here yet (because this is the first pass and import resolving is done in the second pass)

I haven't thought about the exact details how to do this yet, but I don't think this fully does what we want yet.

Reply:
I just looked into dsymbol second.d which uses quite frequently 
example line
second.d: 265
```
currentSymbol = currentSymbol.getFirstPartNamed(part);
```
I think to be more accurate I should be using this function.

Reply:
okay I am not sure either, but I think it would also be nice to make a test case that covers this.